pkg/services/folder/folderimpl/folder_test.go

```diff
@@ -178,22 +178,24 @@ func TestIntegrationFolderService(t *testing.T) {
 
 		t.Run("Given user has permission to save", func(t *testing.T) {
 			origNewGuardian := guardian.New
-			guardian.MockDashboardGuardian(&guardian.FakeDashboardGuardian{CanSaveValue: true})
+			guardian.MockDashboardGuardian(&guardian.FakeDashboardGuardian{CanSaveValue: true, CanViewValue: true})
 			service.features = featuremgmt.WithFeatures()
 
 			t.Run("When creating folder should not return access denied error", func(t *testing.T) {
 				dash := dashboards.NewDashboardFolder("Test-Folder")
 				dash.ID = rand.Int63()
+				dash.UID = util.GenerateShortUID()
 				f := dashboards.FromDashboard(dash)
 
 				dashStore.On("ValidateDashboardBeforeSave", mock.Anything, mock.AnythingOfType("*dashboards.Dashboard"), mock.AnythingOfType("bool")).Return(true, nil)
 				dashStore.On("SaveDashboard", mock.Anything, mock.AnythingOfType("dashboards.SaveDashboardCommand")).Return(dash, nil).Once()
+				dashStore.On("DeleteDashboard", mock.Anything, mock.AnythingOfType("*dashboards.DeleteDashboardCommand")).Return(nil).Once()
```

Comment:
Just to make sure I understood the change properly, could you please point me to the code change which requires this test change?

Also congrats on the PR 🎉

Reply:
Good catch! Thanks for the review.
It's not needed anymore. Initially added it cause with my earlier changes the test kept failing with `I don't know what to return because the method call was unexpected.`

Reply:
Makes sense 👍